compat/src/index.d.ts

```diff
@@ -1,75 +1,77 @@
 import * as _hooks from '../../hooks';
 import * as preact from '../../src';
+import { JSXInternal } from '../../src/jsx'
 import { ForwardFn } from './internal';
-export * from './suspense';
-import { Suspense, lazy } from './suspense';
-
-export * from '../../hooks';
-export import Component = preact.Component;
-export import createContext = preact.createContext;
-export import createRef = preact.createRef;
-export import Fragment = preact.Fragment;
-export import createElement = preact.createElement
-export import cloneElement = preact.cloneElement
-
-export declare const version: string;
-
-export declare function createPortal(vnode: preact.VNode, container: Element | Element): preact.VNode<any>;
-
-export declare function render(vnode: preact.VNode<any>, parent: Element, callback?: () => void): Component | null;
-
-export declare function hydrate(vnode: preact.VNode<any>, parent: Element, callback?: () => void): Component | null;
-
-export declare function unmountComponentAtNode(container: Element | Document | ShadowRoot | DocumentFragment): boolean;
-
-export declare function createFactory(type: preact.VNode["type"]): preact.VNode<{}>;
-
-export declare function isValidElement(element: any): boolean;
-
-export declare function findDOMNode(component: preact.Component): Element | null;
-
-export declare interface PureComponent<P = {}, S = {}> extends preact.Component {
-  isPureReactComponenet: boolean;
+import * as _Suspense from './suspense';
+
+// export default React;
+export = React;
+export as namespace React;
+declare namespace React {
```

Comment:
could this conflict with `@types/react`?

Reply:
It could as they also use the `React` namespace. I'll have to try them with react. I think that it you're using React + TS and using a bundle it will preferentially use the React typings.

I'll test this first thing tomorrow.

Reply:
Tested using my [Preact/TS boilerplate](https://github.com/pmkroeker/pmkroeker.github.io), with installed React + @types/react and its components and it works as intended. The typescript compiler runs before the webpack alias is resolved.

Reply:
right, but if you don't import compat directly, do these typings even do anything?

Reply:
That's correct. Unless you either import from `preact/compat` or use an alias in tsconfig for `react: preact/compat`, these typings will not be used.

Reply:
but what if we do import it 

and have `@types/react` somewhere else, i.e. a dependency that gets aliased

Reply:
I believe it will pick the typings that the dependency uses. With typescript its more difficult to mix and match React / Preact, as you have to specify the jsxFactory in your config. If they are both somehow referenced, the worst case scenario is the namespaces get merged together, which should be fine.

Reply:
I just tested out importing a React Component into a preact/compat component.
The basic pieces worked, but there is a difference in the typings between the 2 in how `componentDidCatch` is typed, which causes a build error.

Reply:
https://github.com/preactjs/preact/blob/a19aee98c64fca277f4ea06527c105f4c5363170/src/diff/index.js#L336 is the difference between the 2 typings. React passes both the error and errorInfo into `componentDidCatch(error, errorInfo)`, where as we just pass error. Adding the second argument to the typings resolved the issue, but possibly should be addressed.

Reply:
well, `jsxFactory` doesn't matter for dependencies, because they're generally compiled

Reply:
True, I don't see any potential issues. If they call React internally they will use the React typings.